src/sagemaker/serve/builder/jumpstart_builder.py

```diff
@@ -718,20 +718,23 @@ def _optimize_for_jumpstart(
                 f"Model '{self.model}' requires accepting end-user license agreement (EULA)."
             )
 
-        is_compilation = (not quantization_config) and (
-            (compilation_config is not None) or _is_inferentia_or_trainium(instance_type)
+        is_compilation = (compilation_config is not None) or _is_inferentia_or_trainium(
+            instance_type
         )
 
         pysdk_model_env_vars = dict()
         if is_compilation:
             pysdk_model_env_vars = self._get_neuron_model_env_vars(instance_type)
 
+        # optimization_config can contain configs for both quantization and compilation
         optimization_config, override_env = _extract_optimization_config_and_env(
             quantization_config, compilation_config
         )
-        if not optimization_config and is_compilation:
+        if (
+            not optimization_config or not optimization_config.get("ModelCompilationConfig")
+        ) and is_compilation:
             override_env = override_env or pysdk_model_env_vars
-            optimization_config = {
+            optimization_config["ModelCompilationConfig"] = {
```

Comment:
we are adding "ModelCompilationConfig" twice in this structure?

Reply:
Whoops good catch. Fixed.